### PR TITLE
Merge from master

### PR DESCRIPTION
**Pull request by ikostan**
## Summary by Sourcery

Add solutions for "Pointless Farmer", "A Strange Trip to the Market", and "Find the in-between point" Kata.

New Features:
- Implement the "buy_or_sell" function to determine the sequence of buy/sell actions for fruit conversions, returning "ERROR" if the original fruit cannot be obtained back.

Tests:
- Add unit tests for the "buy_or_sell", "is_loch_ness_monster", and "middle_point" functions with positive and negative test cases using parameterized inputs and descriptive documentation.

### CODE REVIEW

File: kyu_7/README.md [hunk 1]
```diff
@@ -15,42 +15,42 @@ rank - the harder the kata the faster you advance.
...
-| 27  | [Basic Math (Add or Subtract)](https://www.codewars.com/kata/5809b62808ad92e31b000031)                        |                                                          [Solution](https://github.com/ikostan/codewars/tree/master/kyu_7/basic_math_add_or_subtract) |
-| 28  | [Sum of odd numbers](https://www.codewars.com/kata/55fd2d567d94ac3bc9000064)                                  |                                                                  [Solution](https://github.com/ikostan/codewars/tree/master/kyu_7/sum_of_odd_numbers) |
-| 29  | [Help Bob count letters and digits](https://www.codewars.com/kata/5738f5ea9545204cec000155)                   |                                                   [Solution](https://github.com/ikostan/codewars/tree/master/kyu_7/help_bob_count_letters_and_digits) |
-| 30  | [Make Class](https://www.codewars.com/kata/5d774cfde98179002a7cb3c8)                                          |                                                                          [Solution](https://github.com/ikostan/codewars/tree/master/kyu_7/make_class) |
```
[the remark this answers is not in the document]
Comment (ikostan) [reply kyu_7/README.md:50]: Fixed, now "Make Class" is #17 on the list.
